tinygrad/ops.py

```diff
@@ -26,16 +26,11 @@ class LazyOp:
   src: Tuple[Union[LazyOp, LazyBuffer], ...]
   arg: Any
   buffers: Tuple[LazyBuffer, ...]
-
   def __init__(self, op: Op, src: Tuple[Union[LazyOp, LazyBuffer], ...], arg: Any = None):
-    self.op = op
-    self.src = src
-    self.arg = arg
-    try:
-      self.buffers = tuple([y for x in src for y in x.buffers])
-    except AttributeError:
-      # NOTE: the linearizer's key function maps the buffers to ints, and LOCAL_BUFFER is used. we don't care about buffers in these cases
-      pass
+    self.op, self.src, self.arg, self.buffers = op, src, arg, ()
+    try:  # NOTE: the linearizer's key function maps the buffers to ints, and LOCAL_BUFFER is used. we don't care about buffers in these cases
+      for x in src: self.buffers += x.buffers
```

Comment:
This change behaves differently when an attribute error is thrown. Self.buffers will not be empty.
 Not sure if that would be a problem.

Reply:
Inderdaad. Thanks! https://github.com/tinygrad/tinygrad/pull/1689